From 13257656d294300ba55b6d02798b16d912bc43a1 Mon Sep 17 00:00:00 2001
From: Hyun Kwon <hyun.kwon@xilinx.com>
Date: Tue, 11 Aug 2020 18:38:41 -0700
Subject: [PATCH 1511/1852] misc: xilinx-ai-engine: include stdlib.h for size_t
 in uapi header
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Include stdlib.h under #ifndef __KERNEL__. Otherwise, errors as below:

./usr/include/linux/xlnx-ai-engine.h:56:2: error: unknown type name ‘size_t’
  size_t offset;
  ^
./usr/include/linux/xlnx-ai-engine.h:57:2: error: unknown type name ‘size_t’
  size_t size;
  ^

Reported-by: kernel test robot <lkp@intel.com>
Signed-off-by: Hyun Kwon <hyun.kwon@xilinx.com>
Reviewed-by: Wendy Liang <wendy.liang@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 include/uapi/linux/xlnx-ai-engine.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/include/uapi/linux/xlnx-ai-engine.h b/include/uapi/linux/xlnx-ai-engine.h
index 579f31c4695a..dfc52d9e28be 100644
--- a/include/uapi/linux/xlnx-ai-engine.h
+++ b/include/uapi/linux/xlnx-ai-engine.h
@@ -6,6 +6,10 @@
 #ifndef _UAPI_AI_ENGINE_H_
 #define _UAPI_AI_ENGINE_H_
 
+#ifndef __KERNEL__
+#include <stdlib.h>
+#endif
+
 #include <linux/ioctl.h>
 #include <linux/types.h>
 
-- 
2.31.1

